kuronometer-core/src/main/scala/com/github/pedrovgs/kuronometer/Kuronometer.scala

```diff
@@ -0,0 +1,92 @@
+package com.github.pedrovgs.kuronometer
+
+import cats.free.Free
+import com.github.pedrovgs.kuronometer.KuronometerResults.{ConnectionError, KuronometerResult}
+import com.github.pedrovgs.kuronometer.app.KuronometerProgram
+import com.github.pedrovgs.kuronometer.free.algebra.{ReporterOps, ViewOps}
+import com.github.pedrovgs.kuronometer.free.domain.View.Message
+import com.github.pedrovgs.kuronometer.free.domain._
+import com.github.pedrovgs.kuronometer.free.interpreter.Interpreters
+import com.github.pedrovgs.kuronometer.free.interpreter.formatter.DurationFormatter.NanosecondsFormat.format
+import com.github.pedrovgs.kuronometer.free.interpreter.formatter.SummaryBuildStageExecutionFormatter
+
+class Kuronometer(implicit R: ReporterOps[KuronometerProgram], V: ViewOps[KuronometerProgram], interpreters: Interpreters) {
+
+  private val kuronometerHeader = "== Kuronometer =="
+
+  def reportBuildFinished(buildExecution: BuildExecution, config: Config): KuronometerResult[BuildExecution] = {
+    val filteredBuildExecution = filterBuildExecutionData(buildExecution, config)
+    val program = for {
+      result <- reportBuildExecution(filteredBuildExecution, config)
+      _ <- showReportResult(result, config)
```

Comment:
If you want to run 2 actions discarding the value of the second one https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/FlatMap.scala#L73